types/proposal.go

```diff
@@ -33,21 +34,27 @@ type Proposal struct {
 // If there is no POLRound, polRound should be -1.
 func NewProposal(height int64, round int, blockPartsHeader PartSetHeader, polRound int, polBlockID BlockID) *Proposal {
 	return &Proposal{
+		Type:             ProposalType,
 		Height:           height,
 		Round:            round,
+		POLRound:         polRound,
 		Timestamp:        tmtime.Now(),
 		BlockPartsHeader: blockPartsHeader,
-		POLRound:         polRound,
 		POLBlockID:       polBlockID,
 	}
 }
 
 // String returns a string representation of the Proposal.
 func (p *Proposal) String() string {
-	return fmt.Sprintf("Proposal{%v/%v %v (%v,%v) %X @ %s}",
-		p.Height, p.Round, p.BlockPartsHeader, p.POLRound,
+	return fmt.Sprintf("Proposal{%X(Proposal) %v/%v %v (%v,%v) %X @ %s}",
```

Comment:
I don't think we need the `%X(Proposal)` since we already have `Proposal{`

Reply:
Thought I make it consistent with Vote but you are right. Removed.